staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -334,24 +339,83 @@ func ValidateWebhookConfiguration(fldPath *field.Path, c *api.WebhookConfigurati
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("connectionInfo", "type"), c.ConnectionInfo, []string{api.AuthorizationWebhookConnectionInfoTypeInCluster, api.AuthorizationWebhookConnectionInfoTypeKubeConfigFile}))
 	}
 
-	// TODO: Remove this check and ensure that correct validations below for MatchConditions are added
-	// for i, condition := range c.MatchConditions {
-	//	 fldPath := fldPath.Child("matchConditions").Index(i).Child("expression")
-	//	 if len(strings.TrimSpace(condition.Expression)) == 0 {
-	//	     allErrs = append(allErrs, field.Required(fldPath, ""))
-	//	 } else {
-	//		 allErrs = append(allErrs, ValidateWebhookMatchCondition(fldPath, sampleSAR, condition.Expression)...)
-	//	 }
-	// }
-	if len(c.MatchConditions) != 0 {
-		allErrs = append(allErrs, field.NotSupported(fldPath.Child("matchConditions"), c.MatchConditions, []string{}))
-	}
+	_, errs := validateMatchConditions(c.MatchConditions, fldPath, utilfeature.DefaultFeatureGate.Enabled(features.StructuredAuthorizationConfiguration))
```

Comment:
I don't think we need the feature gate check here or in ValidateMatchConditions... the only way to get a value in in c.MatchConditions is to load from a config file, which is only allowed if the feature is on

this would be like checking the feature gate before allowing more than one webhook authz entry in the config file... the only way to do that is by loading a config file which is guarded by the feature gate earlier.

Reply:
I asked for the feature gate check in validation because people can import the authn/z logic directly and use it (have done it before myself), so this being there forces them to set the feature gate as well before they start using it.

Reply:
hmm... isn't that like any other aspect that is only controllable via the config file? multiple webhooks, decisionOnError, timeout, etc... we're not checking the feature gate for every use of those things

Reply:
I don't feel that strongly, I just think it's inconsistent and not that necessary

Reply:
just to confirm, we should drop feature gate check here but still need to check in webhook.go right? https://github.com/kubernetes/kubernetes/pull/121223/files#diff-8b41b5e2c7d3bb27fa452e58ab0a4120b02d3a7a16d8961954f5c3c270ab8935R115-R117

Reply:
eh, that's a fair point... I probably *would* gate that off since it's a net new code path inside the webhook evaluation function. If we want to keep the gate check here to fail early at construction time instead of at runtime I guess that's ok